ecommerce/serializers.py

```diff
@@ -332,8 +388,20 @@ def validate_coupons(self, coupons):
         """Can't do much validation here since we don't have the product version id"""
         return {"coupons": coupons}
 
+    def validate_data_consents(self, data_consents):
+        """Validate that DataConsentUser objects exist"""
+        invalid_consent_ids = []
+        for consent_id in data_consents:
```

Comment:
Can we rewrite to avoid n+1 queries? Maybe

    valid_consent_ids = set(models.DataConsentUser.objects.filter(id__in=data_consents).values_list("id", flat=True)
    invalid_consent_ids = set(data_consents) - valid_consent_ids